nullaway/src/test/java/com/uber/nullaway/NullAwaySerializationTest.java

```diff
@@ -810,4 +827,65 @@ public void errorSerializationTest() {
         .setOutputFileNameAndHeader(ERROR_FILE_NAME, ERROR_FILE_HEADER)
         .doTest();
   }
+
+  @Test
+  public void fieldInitializationSerializationTest() {
+    Path tempRoot = Paths.get(temporaryFolder.getRoot().getAbsolutePath(), "test_field_init");
+    String output = tempRoot.toString();
+    try {
+      Files.createDirectories(tempRoot);
+      FixSerializationConfig.Builder builder =
+          new FixSerializationConfig.Builder()
+              .setSuggest(true, false)
+              .setFieldInitInfo(true)
+              .setOutputDirectory(output);
+      Path config = tempRoot.resolve("serializer.xml");
+      Files.createFile(config);
+      configPath = config.toString();
+      builder.writeAsXML(configPath);
+    } catch (IOException ex) {
+      throw new UncheckedIOException(ex);
+    }
+    SerializationTestHelper<FieldInitDisplay> tester = new SerializationTestHelper<>(tempRoot);
+    tester
+        .setArgs(
+            Arrays.asList(
+                "-d",
+                temporaryFolder.getRoot().getAbsolutePath(),
+                "-XepOpt:NullAway:AnnotatedPackages=com.uber",
+                "-XepOpt:NullAway:SerializeFixMetadata=true",
+                "-XepOpt:NullAway:FixSerializationConfigPath=" + configPath))
+        .addSourceLines(
+            "com/uber/Test.java",
+            "package com.uber;",
+            "import javax.annotation.Nullable;",
+            "public class Test {",
+            "   Object foo;",
+            "   @Nullable Object nullableFoo;",
+            "   Test() {",
+            "       foo = new Object();",
```

Comment:
I think this test case is weird in that the constructor already initializes `foo`. Wouldn't it make more sense to have an empty constructor and a `// BUG: ` expected error for non-initialization in this code? That would be more properly representative of the code we want this detection/handler to run on, wouldn't it?

Reply:
The intention of initializing `foo` in constructor was to make sure constructors are not selected as candidates for initializers, therefore, I removed the initialization of `foo` and made another field `bar`, and initialized it in the constructor to make the test more understandable. 

03ab00deb874aa00b71cc7e033ccd7a9f10e30dc